infra/main.bicep

```diff
@@ -0,0 +1,161 @@
+        name: embeddingDeploymentName
+        model: {
+          format: 'OpenAI'
+          name: embeddingModelName
+          version: '2'
+        }
+        capacity: embeddingDeploymentCapacity
+      }
+    ]
+    keyVaultName: keyVault.outputs.name
+  }
+}
+// USER ROLES
+module openAiRoleUser 'core/security/role.bicep' = {
+  name: 'openai-role-user'
+  params: {
+    principalId: principalId
+    roleDefinitionId: '5e0bd9bd-7b93-4f28-af87-19fc36ad61bd'
+    principalType: 'User'
+  }
+}
+module dataBase 'app/sqlserver.bicep' ={
+  name: !empty(dbServiceName) ? dbServiceName : '${abbrs.sqlServers}catalog-${resourceToken}'
+  params:{
+    location: location
+    appUserPassword: appUserPassword
+    sqlAdminPassword: sqlAdminPassword
+    databaseName: dbName
+    keyVaultName: keyVault.outputs.name
+    name: !empty(dbServiceName) ? dbServiceName : '${abbrs.sqlServers}catalog-${resourceToken}'
+    openAIUrl: openAi.outputs.endpoint
+    appUser: appUser
+    sqlAdmin: sqlAdmin
+    openAIKey: kv.getSecret('openAlKey')
+  }
+}
+// Store secrets in a keyvault
+module keyVault 'app/keyvault.bicep' = {
+  name: 'keyvault'
+  params: {
+    name: !empty(keyVaultName) ? keyVaultName : '${abbrs.keyVaultVaults}${resourceToken}'
+    location: location
+    tags: tags
+    principalId: principalId
+  }
+}
+module web 'app/staticwebapp.bicep' = {
+  name: 'web'
+  params:{
+    name: staticWebAppName
+    location: location
+    tags: union(tags, { 'azd-service-name': 'web'})
+    repositoryToken: repositoryToken
+    repositoryUrl: repositoryUrl
+    sqlConnectionString: kv.getSecret('AZURE-SQL-CONNECTION-STRING')
+    sqlServerId: dataBase.outputs.id
+    sqlServerLocation: dataBase.outputs.location
+  }
+}
+module hostingPlan 'core/host/appserviceplan.bicep'= {
+  name: 'hostingPlan'
+  params: {
+    location: location
+    name: hostingPlanName
+    sku: {
+      name: 'Y1'
+      tier: 'Dynamic'
+    }
+    reserved: false
+  }
+}
+module applicationInsights 'app/applicationInsights.bicep' = {
+  name: 'applicationInsights'
+  params: {
+    name: applicationInsightsName
+    location: location
+    keyVaultName: keyVault.outputs.name
+  }
+}
+module functionApp 'app/functions.bicep' = {
+  name: 'functionApp'
+  params: {
+    location: location
+    storageAccountName: storageAccount.name
+    instrumentationKey: kv.getSecret('instrumentationKey')
+    openAIEndpoint: openAi.outputs.endpoint
+    openAiKey: kv.getSecret('openAlKey')
+    functionAppName: functionAppName
+    hostingPlanId: hostingPlan.outputs.id
+    storageAccountKey: kv.getSecret('storageAccountKey')
+    sqlConnectionString: kv.getSecret('AZURE-SQL-CONNECTION-STRING')
+  }
+  dependsOn: [dataBase]
+}
+module storageAccount 'app/storageaccount.bicep' = {
+  name: storageAccountName
+  params: {
+    name: storageAccountName
+    location: location
+    keyVaultName: keyVault.outputs.name
+  }
+}
+resource kv 'Microsoft.KeyVault/vaults@2023-02-01' existing = {

```

Comment:
Why do we need this ref?

Reply:
Because I originally wanted to call the kv.getSecret('storageAccountKey') method in the main.bicep file. However, this method of obtaining the key has been rejected through the [PR](https://github.com/jongio/chat-with-your-data-solution-accelerator/pull/2#discussion_r1397602513), and is now being prepared to be modified.

Reply:
Ref key vault to get service keys. Next, we will use `AUTH_TYPE` (RBAC or KEYS) to replace it.